### ISSUE TITLE
Dropdown, Dropdown Group, Dropdown - Ensure all interactive functionality is operable with the keyboard - (2047838630)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5636]
### Module: 
 43 Dropdown, Dropdown Group, Dropdown 

### Violation: 
 Ensure all interactive functionality is operable with the keyboard 

### WCAG Reference: 
- [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) 

### Instance ID: 
 2047838630 

### Severity: 
 8 

### Description: 
 [Issue]
This doesn&#039;t respond to space bar or enter

[User Impact]
All functionality must be actionable regardless of the input method used. Navigation and functionality that is dependent on specific devices such as a mouse may not be available to users with dexterity impairments as well as people without vision or low vision. These users may not be able to use a mouse so content providers should ensure that all functionality can be used from the keyboard.

[Code Reference] 
 ```
<calcite-dropdown placement="bottom-start" scale="m" type="click" calcite-hydrated="">
      <calcite-link slot="dropdown-trigger" appearance="solid" role="presentation" calcite-hydrated="">Link Trigger</calcite-link>
      <calcite-dropdown-group group-title="Link trigger" aria-label="Link trigger" role="group" selection-mode="single" calcite-hydrated="">
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Grassland</calcite-dropdown-item>
        <calcite-dropdown-item active="" aria-checked="true" role="menuitemradio" tabindex="0" calcite-hydrated="">Forest</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Trail</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Park</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Lake</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">Mountain</calcite-dropdown-item>
        <calcite-dropdown-item aria-checked="false" role="menuitemradio" tabindex="0" calcite-hydrated="">River</calcite-dropdown-item>
      </calcite-dropdown-group>
    </calcite-dropdown>
 ``` 

### Note: 
 [Suggestion]
Ensure that everywhere there is a click listener there is a keypress listener. For links it should include 
Enter and Space to open a select as well as choosing the option.

[Compliant Code Example] 
 ```
This doesn&#039;t respond to space bar or enter
 ```
 

### Media Type: 
 Keyboard Accessibility 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Duplicate of #5637.
